mmcv/runner/checkpoint.py

```diff
@@ -277,14 +277,12 @@ def load_from_http(filename, map_location=None, model_dir=None):
         dict or OrderedDict: The loaded checkpoint.
     """
 
-    rank, world_size = get_dist_info()
-    rank = int(os.environ.get('LOCAL_RANK', rank))
+    rank = int(os.environ.get('LOCAL_RANK', get_dist_info()[0]))
     if rank == 0:
-        checkpoint = model_zoo.load_url(filename, model_dir=model_dir)
-    if world_size > 1:
+        checkpoint = model_zoo.load_url(filename, model_dir=model_dir, map_location=map_location)
+    else:
```

Comment:
Please double check this change

Reply:
For the rank-0 worker, `torch.distributed.barrier()` is also needed in distributed training to wait other processes.

Reply:
L272 may also be updated since the docstring is wrong.